src/shared/parameter_v2/config_definitions/test/example.yaml

```diff
@@ -119,24 +119,24 @@ include:
 # The following are example of special attributes that can be applied to any parameter
 #########################################################################################
 # Make a parameter value optional (CPP only)
-- bool:
-    name: example_optional_parameter
-    value: false
-    optional: true
-    description: >
-      Optional parameters will generate with std::optional as their value return type.
-      They can accept std::nullopt as a value. If optional is not specified, it is
-      assumed to be false.
-
-      Will only generate for CPP code.
+#- bool:
```

Comment:
How hard would be to add? I remember having discussion about how to classify it as numeric or enumerated. Have we figured that out yet?

Reply:
I forget what exactly the issue was that I had ran into, I can look into that. So if it's not difficult, we are interested in adding this? I think I discussed this also with @akhilveeraghanta

Reply:
if it's easy to do, it would be good for parameters where we might override a value, so nullopt means don't override. This can be done later if you'd like

Reply:
Done (issue #1889)